Remove left-over parkour mutation references

eltank opened the pull request:
#### Summary
Bugfixes "Remove left-over parkour mutation references"

#### Purpose of change

#50784 removed the parkour mutation/trait, turning it into a proficiency. However, references were left in mods causing occasional test failures, like:
https://github.com/CleverRaven/Cataclysm-DDA/pull/51474/checks?check_run_id=3547489785

#### Describe the solution

Remove all remaining references. For trait groups that used to have parkour and bad_knees, ~cut the probability in half otherwise bad_knees becomes twice as likely.~ replace with GOODCARDIO as suggested by @Maleclypse.

#### Describe alternatives you've considered


#### Testing

It's a CI error, so letting CI validate the fix.

#### Additional context

```
Mods-(~[slow] ~[.])=>     ./tests/cata_test(generic_factory<mutation_branch>::obj(string_id<mutation_branch> const&) const+0x263) [0x491f195]
Mods-(~[slow] ~[.])=>     ./tests/cata_test(Character::set_mutation_unsafe(string_id<mutation_branch> const&)+0x142) [0x48d4346]
Mods-(~[slow] ~[.])=>     ./tests/cata_test(Character::set_mutation(string_id<mutation_branch> const&)+0xe) [0x48d3f94]
Mods-(~[slow] ~[.])=>     ./tests/cata_test(npc::randomize(string_id<npc_class> const&)+0x3b73) [0x4a38219]
```

```
Mods-(~[slow] ~[.])=> (continued from above) ERROR : src/generic_factory.h:442 [const T &generic_factory<mutation_branch>::obj(const string_id<T> &) const [T = mutation_branch]] invalid trait id "PARKOUR"
Mods-(~[slow] ~[.])=> 18:08:20.566 ERROR : src/generic_factory.h:442 [const T &generic_factory<mutation_branch>::obj(const string_id<T> &) const [T = mutation_branch]] invalid trait id "PARKOUR"
```

Maleclypse commented:
On the distributions of traits in NPCs I replaced it with GOODCARDIO for AFS and Magiclysm in the PR Obsoletion Fixes

eltank commented:
> On the distributions of traits in NPCs I replaced it with GOODCARDIO for AFS and Magiclysm in the PR Obsoletion Fixes

Changed this PR to also do the same.
